crates/boundless-market/build.rs

```diff
@@ -222,11 +236,44 @@ fn copy_artifacts() {
                 .unwrap()
                 .trim_start_matches("0x");
 
-            // Write to new file with .hex extension
-            let dest_file = dest_path.join(format!("{contract}.hex"));
-            let mut file = File::create(dest_file).unwrap();
-            file.write_all(bytecode.as_bytes()).unwrap();
+            // Append the contract definition with embedded bytecode
+            rust_content.push_str(&format!(
+                r#"alloy::sol! {{
+    #[sol(rpc, bytecode = "{}")]
+    contract {} {{
+        {}
+    }}
+}}"#,
+                bytecode,
```

Comment:
What if I have a slightly different solc version installed on my machine? Couldn't this lead to different bytecode and trigger a file change?

Reply:
That is true, we do have some constraints in the forge.toml though to decrease this likelihood